packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -59,14 +70,38 @@ export function hasOne(definition?: Object) {
   return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
 }
 
-/**
- * Decorator for hasMany
- * @param definition
- * @returns {(target:any, key:string)}
- */
-export function hasMany(definition?: Object) {
-  const rel = Object.assign({type: RelationType.hasMany}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function hasMany<T extends typeof Entity>(
+  targetModel: T,
+  definition?: Partial<HasManyDefinition>,
+) {
+  // todo(shimks): extract out common logic (such as @property.array) to
+  // @relation
+  return function(target: Object, key: string) {
+    property.array(targetModel)(target, key);
```

Comment:
Does `property.array` check if that design-type of target[key] is `Array`?

Reply:
Yup